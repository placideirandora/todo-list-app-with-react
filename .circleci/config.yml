version: 2.1
jobs:
  build:
    docker:
      - image: circleci/node:10.16.0

    working_directory: ~/todo-list-app-with-react-js

    steps:
      - checkout
      - restore_cache:
          keys:
            - v1-dependencies-{{ checksum "package.json" }}
            - v1-dependencies-

      - run: npm install

      - save_cache:
          paths:
            - node_modules
          key: v1-dependencies-{{ checksum "package.json" }}

      - run: npm run test:coverage
