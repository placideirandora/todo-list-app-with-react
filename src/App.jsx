import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import NavBar from './components/shared/NavBar';
import Home from './Components/Home';
import About from './Components/About';
import Todo from './Components/Todo';
import AddTodo from './Components/AddTodo';
import Footer from './components/shared/Footer';

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <NavBar />
        <Switch>
          <Route exact path="/" component={Home} />
          <Route path="/Todo" component={Todo} />
          <Route path="/AddTodo" component={AddTodo} />
          <Route path="/About" component={About} />
        </Switch>
        <Footer />
      </BrowserRouter>
    );
  }
}

export default App;
