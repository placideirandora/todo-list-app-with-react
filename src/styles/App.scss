@import 'Colors';
@import 'Fonts';
@import 'Body';

.app {
  .description {
    margin-top: 10rem;
    text-align: center;
    font-size: 3rem;
  }
}
