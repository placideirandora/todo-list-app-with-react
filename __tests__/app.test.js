import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as adapter from './config/enzyme.config';
import App from '../src/App';

describe('App Component Wrapper Test Case', () => {
  it('should render the component wrapper without any failure', () => {
    const component = shallow(<App />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
