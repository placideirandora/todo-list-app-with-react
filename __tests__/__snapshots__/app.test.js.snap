// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Wrapper Test Case should render the component wrapper without any failure 1`] = `
<BrowserRouter>
  <Navbar />
  <section
    style={
      Object {
        "height": "78vh",
        "margin": "0 10%",
        "maxHeight": "78vh",
        "overflow": "scroll",
        "width": "80%",
      }
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Todo)",
            "type": [Function],
          }
        }
        path="/Todo"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(AddTodo)",
            "type": [Function],
          }
        }
        path="/AddTodo"
      />
      <Route
        component={[Function]}
        path="/About"
      />
    </Switch>
  </section>
  <Footer />
</BrowserRouter>
`;
